packages/beacon-node/src/chain/blocks/types.ts

```diff
@@ -53,13 +60,19 @@ const BLOBSIDECAR_FIXED_SIZE = 131256;
 export const getBlockInput = {
   blockInputCache: new Map<RootHex, BlockInputCacheType>(),
 
+  prune(): void {
```

Comment:
The pattern in 

```ts
blockInputCache: new Map<RootHex, BlockInputCacheType>(),
```

is extremely odd and dangerous. I did not picked it up on the initial free the blobs PR. This Map is a global variable shared across the entire JS process. There could be multiple instances of BeaconChain, and all of them re-use the same mapping, potentially causing downstream inconsistencies. What's the reason to have state-full variables here? The purpose of this helpers was just to return BlockInput enum variants. State should be attached to the beacon chain class

Reply:
agree that having a cache here is too much for the purpose of this file `types.ts`, which is originally for types only similar to other places

`chain.ts` is a great place to store this cache, `unknownBlockSync` can base on that to avoid having download block based on root unnecessarily, see #6105

Reply:
suggest moving all of the logic here to a new class called `SeenGossipBlockInput` under `src/chain/seenCache` cc @dapplion @g11tech

Reply:
sounds good, will do